packages/ember-application/lib/system/application-instance.js

```diff
@@ -212,6 +298,255 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
+    JavaScript object containing the desired options into methods that require
+    one of these options object:
+
+    ```javascript
+    MyApp.visit("/", { location: "none", rootElement: "#container" });
+    ```
+
+    Not all combinations of the supported options are valid. See the documentation
+    on `Ember.Application#visit` for the supported configurations.
+
+    Internal, experimental or otherwise unstable flags are marked as private.
+
+    @class BootOptions
+    @namespace @Ember.ApplicationInstance
+    @public
+  */
+  BootOptions = function BootOptions(options) {
+    options = options || {};
+
+    /**
+      Provide a specific instance of jQuery. This is useful in conjunction with
+      the `document` option, as it allows you to use a copy of `jQuery` that is
+      appropiately bound to the foriegn `document` (e.g. a jsdom).
+
+      This is highly experimental and support very incomplete at the moment.
+
+      @property jQuery
+      @type Object
+      @default auto-detected
+      @private
+    */
+    this.jQuery = jQuery;
+
+    /**
+      Interactive mode: whether we need to set up event delegation and invoke
+      lifecycle callbacks on Components.
+
+      @property interactive
+      @type boolean
+      @default auto-detected
+      @private
+    */
+    this.interactive = environment.hasDOM;
+
+    /**
+      Run in a full browser environment.
+
+      When this flag is set to `false`, it will disable most browser-specific
+      and interactive features. Specifically:
+
+      * It does not use `jQuery` to append the root view; the `rootElement`
+        (either specified as a subsequent option or on the applicatio itself)
+        must already be an `Element` in the given `document` (as opposed to a
+        string selector).
+
+      * It does not set up an `EventDispatcher`.
+
+      * It does not run any `Component` lifecycle hooks (such as `didInsertElement`).
+
+      * It sets the `location` option to `"none"`. (If you would like to use
+        the location adapter specified in the app's router instead, you can also
+        specify `{ location: null }` to specifically opt-out.)
+
+      @property browser
+      @type boolean
+      @default auto-detected
+      @public
+    */
+    if (options.browser !== undefined) {
+      this.browser = !!options.browser;
+    } else {
+      this.browser = environment.hasDOM;
+    }
+
+    if (!this.browser) {
+      this.jQuery = null;
+      this.interactive = false;
+      this.location = 'none';
+    }
+
+    /**
+      Disable rendering completely.
+
+      When this flag is set to `true`, it will disable the entire rendering
+      pipeline. Essentially, this puts the app into "routing-only" mode. No
+      templates will be rendered, and no Components will be created.
+
+      @property render
+      @type boolean
+      @default true
+      @public
+    */
+    if (options.render !== undefined) {
+      this.render = !!options.render;
```

Comment:
I wonder if these flags should be renamed so it's more obvious that they're booleans. `render` has the problem of typically being used as a verb.

Maybe `isInteractive` and `shouldRender` instead?